src/auth.ts

```diff
@@ -4,18 +4,29 @@ import { Constants, getResourceIdFromPath, HTTPMethod, ResourceType } from "./co
 import { IHeaders } from "./queryExecutionContext";
 
 /** @hidden */
-export interface IRequestInfo {
-  [index: string]: any;
+export interface RequestInfo {
   verb: HTTPMethod;
   path: string;
   resourceId: string;
-  resourceType: string;
+  resourceType: ResourceType;
   headers: IHeaders;
 }
 
-export interface ITokenProvider {
-  getToken: (requestInfo: IRequestInfo, callback?: (err: Error, token: string) => void) => Promise<string>;
-}
+export type TokenProvider = (
```

Comment:
I'm not a fan of having 6 arguments on a callback in JavaScript. It's harder to consume, imo. That's why I had the wrapper object. What's the value in removing the wrapper?

Reply:
I think I did this so the call sig matched the others in the file. I can change back though.

Reply:
The other ones are internal so I didn't touch them from v1, but I added this one which is why it has the "nicer" model. My preference would be to leave it as a small set of arguments with that wrapper object.